Assets/FallingPlatform.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FallingPlatform : MonoBehaviour
+{
+    // Start is called before the first frame update
+
+    public GameObject spriteObject;
+    public float elapsed = 0.0f;
+    public float duration = 1.5f;
+    public float magnitudeX = 0.3f;
+    public float magnitudeY = 0.3f;
+    public float RestoreTime = 5f;
+    Vector3 spriteOriginalPos;
+    Vector3 objectOriginalPos;
+    Rigidbody2D localRB ; 
+
+    void Start()
+    {
+
+        spriteOriginalPos = spriteObject.transform.position;
+        objectOriginalPos = transform.position;
+        localRB = GetComponent<Rigidbody2D>();
+
+    }
+    void Restore()
+    {
+        localRB.isKinematic = true;
+        localRB.constraints = RigidbodyConstraints2D.FreezeAll;
+        transform.position = objectOriginalPos;
+        spriteObject.transform.position = spriteOriginalPos;
+        elapsed = 0.0f; 
+    }
+    void OnCollisionEnter2D(Collision2D other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            StartCoroutine(Shake());
+            Invoke("Fall", duration);
```

Comment:
Why not place both of these actions in the same IEnumerator? Also if we want to split things out, we can yield another IEnumerator inside of an IEnumerator:

```
IEnumerator DoTheThings() {
  performSomeTask();
  yield return newWaitForSeconds(someAmountOfTime);
  yield return OtherTask();
}

IEnumerator OtherTask() {
  doOtherStuff();
  yield return new WaitForEndOfFrame();
}
```

Reply:
Having everything in one coroutine would be less error-prone and simpler to reason about, but if this works we can def keep it as-is

Reply:
Fixed , this does look cleaner after all. 
    IEnumerator OnCollisionEnter2D(Collision2D other)
    {
        if (other.gameObject.CompareTag("Player"))
        {
            StartCoroutine(Shake());
            yield return new WaitForSeconds(duration);
            Fall();
            yield return new WaitForSeconds(RestoreTime);
            Restore();
        }
    }